CRM/Export/BAO/ExportProcessor.php

```diff
@@ -135,6 +135,37 @@ public function setIsMergeSameAddress(bool $isMergeSameAddress) {
     $this->isMergeSameAddress = $isMergeSameAddress;
   }
 
+  /**
+   * Additional fields required to export postal fields.
+   *
+   * @var array
+   */
+  protected $additionalFieldsForPostalExport = [];
+
+  /**
+   * Get additional fields required to do a postal export.
+   *
+   * @return array
+   */
+  public function getAdditionalFieldsForPostalExport() {
+    return $this->additionalFieldsForPostalExport;
+  }
+
+  /**
+   * Set additional fields required for a postal export.
+   */
+  public function setAdditionalFieldsForPostalExport() {
+    if ($this->getRequestedFields() && $this->isPostalableOnly()) {
+      $fields = ['is_deceased', 'do_not_mail', 'street_address', 'supplemental_address_1'];
+      foreach ($fields as $index => $field) {
+        if (!empty($this->getReturnProperties()[$field])) {
+          unset($field[$index]);
```

Comment:
@eileenmcnaughton this seems different to the previous code also the previous code seems to ensure that `$this->getReturnProperties()` gets added if the field is added to that array if we can't find it already

Reply:
Right the array of additionalFieldsForPostalExport is now merged in when we retrieve it

https://github.com/civicrm/civicrm-core/pull/14790/files#diff-ab4b10ab1ddedcbcc3be13fb06348789R347

So we can get it separately for some purpose but also get the full set